### PR TITLE
[skip release][Breaking]Support v2 API key

### PR DESCRIPTION
**Pull request by AsiaCao**
The LandingLens platform released a new version of the API key (i.e. v2). This PR supports the new API key v2.
(The current v1 key requires the end users to copy two strings which creates friction, the v2 key only has one key string.)

**Main changes**

1. Introduced a new `APIKey` class
2. [**Breaking**]Removed the `api_secret` parameter in `Predictor` class
3. [**Breaking**]Made the `api_key` parameter as a named parameter.
4. Updated `_load_api_credential()` to support both API key v1 and v2.

After this PR, API key v1 can only be configured via env vars or `.env` file, i.e. it can't be passed to Predictor via constructor.

## Migration Guide

Below section shows you how to fix the backward incompatible changes when you upgrade the version to `0.1.0`.

1. Generate your v2 API key from LandingLens. See [here](https://support.landing.ai/docs/api-key) for more information.
2. The `api_secret` parameter is removed in the `Predictor` and `OcrPredictor` class. `api_key` is a named parameter now, which means you must specify the parameter name, i.e. `api_key` if you want to pass it `Predictor` as an argument.  See below code as an example:
    **Before upgrade to `0.1.0`**
    ```
    predictor = Predictor(endpoint_id, api_key, api_secret)
    ```
    **After upgrade to `0.1.0`**
    ```
    predictor = Predictor(endpoint_id, api_key=api_key)
    ```

### CODE REVIEW

**Review by andy-ldai — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build-backend = "poetry.core.masonry.api"
 
 [tool.poetry]
 name = "landingai"
-version = "0.0.50"
+version = "1.0.0"
```
Comment (cat101) [inline pyproject.toml:7]: I would stay on v0 for longer to not give the impression that the library is production ready... how about v0.1.0?

File: docs/user_guide/2_credentials.md [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
 ```
-   LANDINGAI_API_KEY=v7b0hdyfj6271xy2o9lmiwkkcb12345
-   LANDINGAI_API_SECRET=ao6yjcju7q1e6u0udgwrgknhrx6m4n1o48z81jy6huc059gne047l4fq312345
-```
\ No newline at end of file
+
+### Legacy API key and secret
+
+In the past, LandingLens supports generating a key and secret pair, which is known as API key v1. This approach is not recommended anymore.
+But if you have a v1 key and secret pair, you can still use the Python SDK via either setting environment variables or an `.env` file, i.e. the #2 and #3 methods above.
```
Comment (cat101) [inline docs/user_guide/2_credentials.md:18]: I think LE will take a while to transition (let's talk to andy)... if this is the case we may need a "nicer" approach to run with LE... maybe this is ok and we use on the example code....
```
    os.environ["landingai_api_key"] = _API_KEY
    os.environ["landingai_api_secret"] = _API_SECRET
```

File: landingai/common.py [hunk 3]
```diff
@@ -28,6 +37,37 @@ class Config:
...
+    Environment variables will always take priority over values loaded from a dotenv file.
+    """
+
+    api_key: str
+
+    @validator("api_key")
+    def is_api_key_valid(cls, key: str) -> str:
+        """Check if the API key is a v2 key."""
+        if not key.startswith("land_sk_"):
+            raise InvalidApiKeyError(
```
Comment (cat101) [inline landingai/common.py:60]: Nice check!

**Review by cat101 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (AsiaCao) [reply pyproject.toml:7]: done

**Pull request merged**